src/Nethermind/Nethermind.JsonRpc/Modules/Parity/PeerInfo.cs

```diff
@@ -0,0 +1,93 @@
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Collections.Generic;
+using Nethermind.Blockchain.Synchronization;
+using Nethermind.Network;
+using Nethermind.Network.P2P;
+using Nethermind.Stats.Model;
+using Newtonsoft.Json;
+
+namespace Nethermind.JsonRpc.Modules.Parity
+{
+    public class PeerInfo
+    {
+        [JsonProperty("id", Order = 0)]
+        public string Id { get; set; }
+        
+        [JsonProperty("name", Order = 1)]
+        public string Name { get; set; }
+        
+        [JsonProperty("caps", Order = 2)]
+        public List<string> Caps { get; set; }
+        
+        [JsonProperty("network", Order = 3)]
+        public PeerNetworkInfo Network { get; set; }
+        
+        [JsonProperty("protocols", Order = 4)]
+        public Dictionary<string, EthProtocolInfo> Protocols { get; set; }
+        
+        public PeerInfo(Peer peer)
+        {
+            ISession session = peer.InSession ?? peer.OutSession;
+            PeerNetworkInfo peerNetworkInfo = new PeerNetworkInfo();
+            EthProtocolInfo ethProtocolInfo = new EthProtocolInfo();
+            List<Capability> capabilities = new List<Capability>();
+            Caps = new List<string>();
+
+            if (peer.Node != null)
+            {
+                Name = peer.Node.ClientId;
+                peerNetworkInfo.LocalAddress = peer.Node.Host;
+            }
+            
+            if (session != null)
+            {
+                Id = session.RemoteNodeId.ToString();
+                peerNetworkInfo.RemoteAddress = session.State != SessionState.New ? session.RemoteHost : "Handshake";
+                
+                if (session.TryGetProtocolHandler(Protocol.Eth, out var handler))
+                {
+                    ethProtocolInfo.Version = handler.ProtocolVersion;
+                    if (handler is ISyncPeer syncPeer)
+                    {
+                        ethProtocolInfo.Difficulty = syncPeer.TotalDifficulty;
+                        ethProtocolInfo.HeadHash = syncPeer.HeadHash;
+                    }
+                }
+                
+                if (session.TryGetProtocolHandler(Protocol.P2P, out var p2PHandler))
+                {
+                    if (p2PHandler is P2PProtocolHandler p2PProtocolHandler)
+                    {
+                        capabilities = p2PProtocolHandler.AgreedCapabilities;
+                    }
+                }
+            }
+
+            foreach (Capability capability in capabilities)
+            {
+                Caps.Add(capability.ProtocolCode + "/" + capability.Version);
+            }
+            
+            Network = peerNetworkInfo;
+            
+            Protocols = new Dictionary<string, EthProtocolInfo>();
+            Protocols.Add("eth", ethProtocolInfo);
```

Comment:
how about other protocols?

Reply:
parity doesn't support it, we can but didn't for now